jax/_src/numpy/util.py

```diff
@@ -311,6 +311,22 @@ def promote_dtypes_complex(*args: ArrayLike) -> list[Array]:
           for x in args]
 
 
+def promote_dtypes_integral_default(*args: ArrayLike) -> list[Array]:
+  """Convenience function to apply default promotion to integral accumulators.
+
+  Promotes arguments to their corresponding default integral type, or returns
+  the arguments unchanged."""
+  def _promote(x: ArrayLike) -> Array:
+    x = lax.asarray(x)
+    kind = x.dtype.kind
+    if kind in {'i', 'u'}:
+      default_dtype = dtypes.dtype(dtypes._default_types[kind])
+      if x.dtype.itemsize*8 < default_dtype.itemsize:
+        return x.astype(default_dtype)
+    return x
+  return [_promote(x) for x in args]
```

Comment:
Sorry, when suggesting calling this `promote_dtypes_*`, I may have misunderstood the intent of the function. The `promote_dtypes_*` functions promote all inputs to a common dtype, whereas this one seems to be promoting each input individually.

I think this function should be something like (1) determine the common type of all inputspromote all inputs to a common type, (2) if the type is an integer, then upcast to the default, (3) convert all inputs to this common type. What do you think?

Reply:
Turns out it's a bit of a moot point since this promotion behavior **only** happens in reductions, which means that JAX already handles this correctly.